core/src/main/java/com/scalar/db/transaction/consensuscommit/ParallelExecutor.java

```diff
@@ -180,94 +181,92 @@ private void executeTasksInParallel(
 
     CompletionService<Void> completionService =
         new ExecutorCompletionService<>(parallelExecutorService);
-    tasks.forEach(
-        t ->
-            completionService.submit(
-                () -> {
-                  try {
-                    t.run();
-                  } catch (Exception e) {
-                    logger.warn(
-                        "Failed to run a {} task. Transaction ID: {}", taskName, transactionId, e);
-                    throw e;
-                  }
-                  return null;
-                }));
-
-    if (!noWait) {
-      Exception exception = null;
-      for (int i = 0; i < tasks.size(); i++) {
-        Future<Void> future = ScalarDbUtils.takeUninterruptibly(completionService);
 
-        try {
-          Uninterruptibles.getUninterruptibly(future);
-        } catch (java.util.concurrent.ExecutionException e) {
-          if (e.getCause() instanceof ExecutionException) {
-            if (!stopOnError) {
-              exception = (ExecutionException) e.getCause();
-            } else {
-              throw (ExecutionException) e.getCause();
-            }
-          } else if (e.getCause() instanceof ValidationConflictException) {
-            if (!stopOnError) {
-              exception = (ValidationConflictException) e.getCause();
-            } else {
-              throw (ValidationConflictException) e.getCause();
-            }
-          } else if (e.getCause() instanceof CrudException) {
-            if (!stopOnError) {
-              exception = (CrudException) e.getCause();
-            } else {
-              throw (CrudException) e.getCause();
+    // Submit tasks
+    for (ParallelExecutorTask task : tasks) {
+      completionService.submit(
+          () -> {
+            try {
+              task.run();
+            } catch (Exception e) {
+              logger.warn(
+                  "Failed to run a {} task. Transaction ID: {}", taskName, transactionId, e);
```

Comment:
We need to keep this log because, even after using `addSuppressed()`, some exceptions may not be included as suppressed when `stopOnError` is set to true.